source/objects/responseClass.m

```diff
@@ -462,8 +462,9 @@ function saveViz(obj,simu,body,waves,options)
             % Read in data for each body
             for ibod=1:length(obj.bodies)
                 % Read and assign geometry data
-                bodyMesh(ibod).Points = body(ibod).geometry.vertex;
-                bodyMesh(ibod).Conns = body(ibod).geometry.faces;                
+                readBodyMesh = stlread(body(ibod).geometryFile);
```

Comment:
Alternate option:
   
                if isempty(body(ibod).geometry.vertex)
                    % Import body geometry for linear hydro cases
                    body(ibod).importBodyGeometry(simu.domainSize);
                end
                bodyMesh(ibod).Points = body(ibod).geometry.vertex;
                bodyMesh(ibod).Conns = body(ibod).geometry.face;

Reply:
Thanks for the suggestion @akeeste! I do think it may be useful to import geometry for all cases to have those checks. We can discuss further with the team in our next meeting.